crypto/evp_extra/evp_asn1.c

```diff
@@ -91,6 +100,19 @@ static const EVP_PKEY_ASN1_METHOD *parse_key_type(CBS *cbs) {
     return &rsa_asn1_meth;
   }
 
+#ifdef ENABLE_DILITHIUM
+  // if |cbs| is empty after parsing |oid| from it, we overwrite the contents
```

Comment:
we should do this only if the oid corresponds to one of the pqdsa algorithms. In other words,
```
if (PQDSA_find_asn1_by_nid(OBJ_cbs2nid(&oid)) != NULL) {
  if (CBS_len(cbs) == 0) { ...
}
```

Reply:
Thanks, you're right [369080b](https://github.com/aws/aws-lc/pull/1963/commits/369080b5a595f1880b4edc75a12eb8dcbe643c02)